src/codeu/chat/server/Server.java

```diff
@@ -15,295 +15,424 @@
+        }
 
-    Message message = model.messageById().first(relayMessage.id());
+        Message message = model.messageById().first(relayMessage.id());
 
-    if (message == null) {
-      message = controller.newMessage(relayMessage.id(),
-                                      user.id,
-                                      conversation.id,
-                                      relayMessage.text(),
-                                      relayMessage.time());
+        if (message == null) {
+            message = controller.newMessage(relayMessage.id(),
+                    user.id,
+                    conversation.id,
+                    relayMessage.text(),
+                    relayMessage.time());
+        }
     }
-  }
 
-  private Runnable createSendToRelayEvent(final Uuid userId,
-                                          final Uuid conversationId,
-                                          final Uuid messageId) {
-    return new Runnable() {
-      @Override
-      public void run() {
-        final User user = view.findUser(userId);
-        final ConversationHeader conversation = view.findConversation(conversationId);
-        final Message message = view.findMessage(messageId);
-        relay.write(id,
-                    secret,
-                    relay.pack(user.id, user.name, user.creation),
-                    relay.pack(conversation.id, conversation.title, conversation.creation),
-                    relay.pack(message.id, message.content, message.creation));
-      }
-    };
-  }
+    private Runnable createSendToRelayEvent(final Uuid userId,
+                                            final Uuid conversationId,
+                                            final Uuid messageId) {
+        return new Runnable() {
+            @Override
+            public void run() {
+                final User user = view.findUser(userId);
+                final ConversationHeader conversation = view.findConversation(conversationId);
+                final Message message = view.findMessage(messageId);
+                relay.write(id,
+                        secret,
+                        relay.pack(user.id, user.name, user.creation),
+                        relay.pack(conversation.id, conversation.title, conversation.creation),
+                        relay.pack(message.id, message.content, message.creation));
+            }
+        };
+    }
+
+    public void readTransactionLog() {
+
+        try {
+            File transactionLog = new File("transaction_log.txt");
+            if (!transactionLog.exists()) {
+                transactionLog.createNewFile();
+            }
+            Scanner scan = new Scanner(transactionLog);   //read transaction log if it already exists
+            while (scan.hasNextLine()) {
+                String item = scan.nextLine().trim(); //item in transaction log
+                Tokenizer tokenizer = new Tokenizer(item);
+                String action = tokenizer.next();
+
```

Comment:
I am staring at this function, trying to figure out how issue https://github.com/lysanderp41/CodeUSummer2017/issues/46 is happening.

I'm stumped for tonight, but maybe I'll see it tomorrow. I may be looking in the wrong place, of course.